### PR TITLE
Introduce enable for user store config in attribute mapping section

### PR DESCRIPTION
**Pull request by PasinduYeshan**
### Purpose
- Introduce enable for user store config in attribute mapping section for multi emails and multi mobile related local claims.
- Make "Attribute Mappings" section visible in cloud environment.

Screenshot 2024-11-14 at 20 44 06

Screenshot 2024-11-14 at 20 43 36

### Related Issues
- https://github.com/wso2/product-is/issues/19991

### Related PRs
- https://github.com/wso2/identity-apps/pull/6563
- https://github.com/wso2/identity-apps/pull/6583

### Checklist

- [x] e2e cypress tests locally verified. (for internal contributers)
- [x] Manual test round performed and verified.
- [x] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [x] Relevant backend changes deployed and verified
- [ ] [Unit tests](/docs/testing/UNIT_TESTING.md) provided. (Add links if there are any)
- [ ] [Integration tests](https://github.com/wso2/product-is/tree/master/modules/integration) provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines
- [x] Ran FindSecurityBugs plugin and verified report
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets

### CODE REVIEW

File: features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx [hunk 1]
```diff
@@ -61,7 +75,7 @@ export const EditMappedAttributesLocalClaims: FunctionComponent<EditMappedAttrib
     const {
         claim,
         update,
-        [ "data-testid" ]: testId
+        [ "data-componentid" ]: componentId
```
[the remark this answers is not in the document]
Comment (PasinduYeshan) [reply features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx:78]: I checked, this section is currently hidden in cloud env.

File: features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx [hunk 2]
```diff
@@ -73,39 +87,77 @@ export const EditMappedAttributesLocalClaims: FunctionComponent<EditMappedAttrib
 
     const { t } = useTranslation();
 
-    const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
     const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
+    const hiddenUserStores: string[] = useSelector((state: AppState) => state.config.ui.hiddenUserStores);
```
[the remark this answers is not in the document]
Comment (PasinduYeshan) [reply features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx]: Fixed

File: features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx [hunk 3]
```diff
@@ -46,6 +53,13 @@ interface EditMappedAttributesLocalClaimsPropsInterface extends TestableComponen
     update: () => void;
 }
 
+const USER_STORE_CONFIG_SUPPORTED_CLAIMS: string[] = [
```
[the remark this answers is not in the document]
Comment (PasinduYeshan) [reply features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx]: Moved

File: features/admin.claims.v1/components/edit/local-claim/edit-mapped-attribute-local-claims.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.userstore-acordion {
+    margin: 2 0 2 0;
+}
```
[the remark this answers is not in the document]
Comment (PasinduYeshan) [reply features/admin.claims.v1/components/edit/local-claim/edit-mapped-attribute-local-claims.scss]: This file was accidentaly pushed.

File: features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx [hunk 5]
```diff
@@ -162,39 +220,83 @@ export const EditMappedAttributesLocalClaims: FunctionComponent<EditMappedAttrib
...
+                                                                                        (name: string) =>
+                                                                                            name !== store.name)
+                                                                                );
+                                                                            } else {
+                                                                                setExcludedUserStores([
+                                                                                    ...excludedUserStores,
+                                                                                    store.name
+                                                                                ]);
+                                                                            }
+                                                                        } }
```
Comment (JayaShakthi97) [inline features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx:284]: Shall we consider separating this out to a function?

Comment (PasinduYeshan) [reply features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx]: Moved

File: features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx [hunk 6]
```diff
@@ -162,39 +220,83 @@ export const EditMappedAttributesLocalClaims: FunctionComponent<EditMappedAttrib
...
-                                                    readOnly={ isReadOnly }
-                                                />
-                                            </Grid.Column>
-                                        </Grid.Row>
-                                    );
-                                }) }
-                            </Grid>
+                            { userStore.map((store: UserStoreListItem, index: number) => {
+                                return (
+                                    <>
```
[the remark this answers is not in the document]
Comment (PasinduYeshan) [reply features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx]: Removed

File: features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx [hunk 7]
```diff
@@ -61,7 +69,7 @@ export const EditMappedAttributesLocalClaims: FunctionComponent<EditMappedAttrib
     const {
         claim,
         update,
-        [ "data-testid" ]: testId
+        [ "data-componentid" ]: componentId
```
[the remark this answers is not in the document]
Comment (PasinduYeshan) [reply features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx:73]: Removed

File: features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx [hunk 8]
```diff
@@ -15,27 +15,35 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import { Show } from "@wso2is/access-control";
+import { AccordionDetails } from "@mui/material";
```
[the remark this answers is not in the document]
Comment (PasinduYeshan) [reply features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx]: Changed

File: features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx [hunk 9]
```diff
@@ -73,39 +81,77 @@ export const EditMappedAttributesLocalClaims: FunctionComponent<EditMappedAttrib
...
-                userstore.push(...response.data);
+                if (hiddenUserStores && hiddenUserStores.length > 0) {
+                    response.data.map((store: UserStoreListItem) => {
+                        if (!hiddenUserStores.includes(store?.name)) {
+                            userstore.push(store);
+                        }
+                    });
+                } else {
+                    userstore.push(...response.data);
+                }
```
[the remark this answers is not in the document]
Comment (PasinduYeshan) [reply features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx]: Changed

File: features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx [hunk 10]
```diff
@@ -224,5 +326,5 @@ export const EditMappedAttributesLocalClaims: FunctionComponent<EditMappedAttrib
  * Default props for the component.
  */
 EditMappedAttributesLocalClaims.defaultProps = {
-    "data-testid": "edit-local-claims-mapped-attributes"
+    "data-componentid": "edit-local-claims-mapped-attributes"
 };
```
[the remark this answers is not in the document]
Comment (PasinduYeshan) [reply features/admin.claims.v1/components/edit/local-claim/edit-mapped-attributes-local-claims.tsx]: Removed

**Review by dasuni-30 — DISMISSED** [review]

**Review by JayaShakthi97 — APPROVED** [review]

**Pull request merged**